contributing.md

```diff
@@ -0,0 +1,70 @@
+ 3. **Fork** the repository
+	- A fork creates a copy of the code on your Github, so you can work on it separately from everyone else
+	- You can learn more about forking [here](https://docs.github.com/en/get-started/quickstart/fork-a-repo)
+ 4. Ensure that you have **commit signing** enabled
+	- This ensures that the code you submit was committed by you and not someone else who claims to be you
+	- You can learn more about how to setup commit signing [here](https://www.freecodecamp.org/news/what-is-commit-signing-in-git/ "https://www.freecodecamp.org/news/what-is-commit-signing-in-git/")
+	- If you have already made some commits that you wish to put in a pull request without signing them, then you can follow [this guide](https://dev.to/jmarhee/signing-existing-commits-with-gpg-5b58) on how to fix that.
+ 5. **Clone** the repository to your local computer.
+	- This puts a copy of your fork on your computer so you can edit it
+	- You can learn more about cloning repositories [here](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+ 6. **Build** the project
+	- For a detailed look on how to build {{project-name}} look at our [README file](<./README>)
+ 7. **Start writing** your code
+	- Open up your favorite code editor and make the changes that you wanted to make to the repository
+	- Make sure to test your code with the test command found in our [README file](<./README>)
+ 8. **Write tests** for your code
+	- If you are adding a new feature you should write tests that ensure that if someone make changes to the code it cannot break your new feature without breaking the test
+	- If your code adds a new feature you should also write at least one documentation test. The documentation test's purpose is to demonstrate and document how to use the API feature.
+	- If your code fixes a bug you should write tests that ensure that if someone makes code changes in the future the bug does not re-emerge without breaking test
+	- Please create integration tests, if the addition is large enough to warrant them, and unit tests
+		- Unit tests are tests that ensure the functionality of a single function or small section of code
+		- Integration tests test large large sections of code
+		- Read more about the differences [here](https://www.geeksforgeeks.org/difference-between-unit-testing-and-integration-testing/)
+	- Tests should be put inside the test folder inside the source folder and in a test file relevant to the project you are working in
+	- For more information on how to write tests look at [this link](https://doc.rust-lang.org/book/ch11-01-writing-tests.html)
+ 9. Ensure that the code that you made follows our Rust **coding guidelines**
+	- You can find our list of Rust guidelines [here](./coding-guide) This is a courtesy to the programmers that come after you. The easier your code is to read, the easier it will be for the next person to make modifications.
```

Comment:
while we our working on ours, can we link to https://rust-lang.github.io/api-guidelines/about.html?